docs/library/tables-and-data-grids/ag_grid.md

```diff
@@ -561,3 +561,144 @@ The following props are available for `column_defs` as well as many others that
+
+```python demo exec
+import reflex as rx
+from reflex_ag_grid import ag_grid
+import pandas as pd
+
+df = pd.read_csv(
+    "https://raw.githubusercontent.com/plotly/datasets/master/gapminder2007.csv"
+)
+
+column_defs = [
+    ag_grid.column_def(field="country", checkbox_selection=True),
+    ag_grid.column_def(field="pop"),
+    ag_grid.column_def(field="continent"),
+]
+
+def ag_grid_api_simple():
+    my_api = ag_grid.api(id="ag_grid_basic_row_selection")
+    return rx.vstack(
+            ag_grid(
+            id="ag_grid_basic_row_selection",
+            row_data=df.to_dict("records"),
+            column_defs=column_defs,
+        ),
+        rx.button("Select All", on_click=my_api.select_all()),
+        rx.button("Deselect All", on_click=my_api.deselect_all()),
+        spacing="4",
+    )
+```
+
+
+The react code for the `select_all()` event handler is `selectAll = (source?: SelectionEventSourceType) => void;`. 
+
+To use this in Reflex as you can see, it should be called in snake case rather than camel case, and as it returns `void` this means that the function should be called with no arguments. 
+
+
+
+```md alert info
+# Another way to use the AG Grid API
+It is also possible to use the AG Grid API directly with the event trigger (`on_click`) of the component. This removes the need to create a variable `my_api`. This is shown in the example below. It is necessary to use the `id` of the `ag_grid` component that is to be referenced.
+```python
+rx.button("Select all", on_click=ag_grid.api(id="ag_grid_basic_row_selection").select_all()),
+```
+
+
+### More examples
+
+The following example lets a user [export the data as a csv](https://www.ag-grid.com/javascript-data-grid/grid-api/#reference-export-exportDataAsCsv) and [adjust the size of columns to fit the available horizontal space](https://www.ag-grid.com/javascript-data-grid/grid-api/#reference-columnSizing-sizeColumnsToFit). (Try resizing the screen and then clicking the resize columns button)
+
+
+```python demo exec
+import reflex as rx
+from reflex_ag_grid import ag_grid
+import pandas as pd
+
+df = pd.read_csv(
+    "https://raw.githubusercontent.com/plotly/datasets/master/gapminder2007.csv"
+)
+
+column_defs = [
+    ag_grid.column_def(field="country", checkbox_selection=True),
+    ag_grid.column_def(field="pop"),
+    ag_grid.column_def(field="continent"),
+]
+
+def ag_grid_api_simple2():
+    my_api = ag_grid.api(id="ag_grid_export_and_resize")
+    return rx.vstack(
+            ag_grid(
+            id="ag_grid_export_and_resize",
+            row_data=df.to_dict("records"),
+            column_defs=column_defs,
+        ),
+        rx.button("Export", on_click=my_api.export_data_as_csv()),
+        rx.button("Resize Columns", on_click=my_api.size_columns_to_fit()),
+        spacing="4",
+    )
+```
+
+The react code for both of these is shown below. The key point to see is that both of these functions return `void` and therefore should be called with no arguments.
```

Comment:
again here, the `void` is for the return. the arguments accepted are listed in the parentheses

Reply:
fixed